src/main/java/com/farion/onlinebookstore/entity/User.java

```diff
@@ -1,28 +1,97 @@
 package com.farion.onlinebookstore.entity;
 
+import static com.farion.onlinebookstore.entity.Role.RoleName.ROLE_ADMIN;
+import static com.farion.onlinebookstore.entity.Role.RoleName.ROLE_USER;
+
 import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
 import jakarta.persistence.Table;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Data
 @Entity
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
 @Table(name = "users")
-public class User {
+public class User implements UserDetails {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     @Column(nullable = false, unique = true)
     private String email;
     @Column(nullable = false)
     private String password;
-    @Column(name = "first_name", nullable = false)
+    @Column(nullable = false)
     private String firstName;
-    @Column(name = "last_name", nullable = false)
+    @Column(nullable = false)
     private String lastName;
-    @Column(name = "shipping_address")
     private String shippingAddress;
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        Set<SimpleGrantedAuthority> authorities = new HashSet<>();
+        for (Role role : roles) {
+            switch (role.getName()) {
+                case ROLE_ADMIN -> authorities.add(new SimpleGrantedAuthority(ROLE_ADMIN.name()));
+                case ROLE_USER -> authorities.add(new SimpleGrantedAuthority(ROLE_USER.name()));
+                default ->
+                        throw new IllegalArgumentException("Unsupported role: " + role.getName());
+            }
```

Comment:
switch looks redundant
```suggestion
            authorities.add(new SimpleGrantedAuthority(role.name()));
```